src/Accounts/Authenticators.NetFx/DesktopAuthenticatiorBuilder.cs

```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.Azure.Commands.Common.Authentication;
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using System;
+
+namespace Microsoft.Azure.PowerShell.Authenticators.NetFramework
+{
+    public class DesktopAuthenticatorBuilder : IAuthenticatorBuilder
+    {
+        public IAuthenticator Authenticator { get; set; }
+
+        public static void Apply(IAzureSession session)
+        {
+            session.RegisterComponent(AuthenticatorBuilder.AuthenticatorBuilderKey, () =>
+            {
+                var userPassword = new UsernamePasswordAuthenticator();
+                userPassword.Next = new InteractiveUserAuthenticator();
```

Comment:
@markcowl nit: should we add the `AppendAuthenticator` method to the `DelegatingAuthenticator` class? This would allow us to call `userPassword.AppendAuthenticator(new InteractiveUserAuthenticator())` and repeat this procedure if we ever decide to allow another form of authentication after these two in this pipeline. We could then call `userPassword.AppendAuthenticator(new FooBarAuthenticator())` rather than instantiate a new object for the `InteractiveUserAuthenticator` class, and call `interactiveUser.Next = new FooBarAuthenticator()`

Reply:
So, I think maybe this could be an extension class that we add as part of this PR - I want to be able to use the IAuthenticatorBuilder abstract factory, but we could definitely add this as an extension to the interface.